src/libs/ReportUtils.ts

```diff
@@ -5134,6 +5152,28 @@ function buildOptimisticTaskReport(
     };
 }
 
+// Manual export
+function buildOptimisticExportIntegrationAction(label = '', markedManually = false): OptimisticExportAction {
+    return {
+        reportActionID: NumberUtils.rand64(),
+        actionName: CONST.REPORT.ACTIONS.TYPE.EXPORTED_TO_INTEGRATION,
+        pendingAction: CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD,
+        actorAccountID: currentUserAccountID,
+        message: [],
+        person: [],
+        automatic: false,
+        avatar: getCurrentUserAvatar(),
+        created: DateUtils.getDBTime(),
+        shouldShow: true,
+        originalMessage: {
+            automaticAction: false,
+            label,
+            lastModified: DateUtils.getDBTime(),
+            markedManually,
+        },
+    };
+}
+
```

Comment:
Hey, @arosiclair could you have a look and confirm that this is the only action that I need to add when offline? The issue said that, but the backend sends a few more Onyx updates. 

If this is the only action, do you know if it has all the required fields?

I managed to reverse-engineer it from the backend response, but it would be great to have confirmation.

Reply:
Yeah this should be the only new action added and the data looks good so far. 

I updated the issue [here](https://github.com/Expensify/App/issues/43040#issuecomment-2209496874) btw. We'll still add this optimistic action, but the text needs to be different while it's optimistic. We should be able to control that [here](https://github.com/Expensify/App/blob/c1ac6f8e4d504bce12df7b1fe63ce10ff211b703/src/libs/ReportActionsUtils.ts#L1257).

Reply:
Just want to make sure: we want to leave this action in the **old** dot report actions section?

Reply:
Actually, I have to work on the EXPORTINTEGRATION action in [this PR](https://github.com/Expensify/App/pull/44930/files) first. So we should just control the copy in the new `ExportIntegration` component.

Reply:
Should this issue be held on https://github.com/Expensify/App/pull/44930/ as well?